File: examples/rule110.go
package main
import ("fmt"; "reflect"; "time"); var _ = []any{reflect.Int, fmt.Print, time.Second}
func main() {
var automataLength = 110


var dead = " "
var alive = "1"


var initArray = dead
for i := 0; i < automataLength-2; i++ {
initArray=initArray+dead
}

initArray=initArray+alive
initArray=initArray+dead
fmt.Print(initArray,"\n")


var deadByte = dead[0]
var aliveByte = alive[0]


for i := 0; i < automataLength-2; i++ {
var newArray = dead

for pos := 0; pos < automataLength-1; pos++ {
var analyze = []any{initArray[pos],initArray[pos+1],initArray[pos+2]}
if analyze==[]any{deadByte,deadByte,deadByte} {
newArray=newArray+dead
} else if analyze==[]any{aliveByte,deadByte,deadByte} {
newArray=newArray+dead
} else if analyze==[]any{aliveByte,aliveByte,aliveByte} {
newArray=newArray+dead
} else {
newArray=newArray+alive
}
}
newArray=newArray+dead
initArray=newArray
fmt.Print(initArray,"\n")
}

}

File: examples/rule110.rs
#[allow(unused_variables)]
fn main() {
    let automata_length = 110;
    let dead = " ";
    let alive = "1";

    let mut init_array = vec![dead];
    for i in 0..automata_length {
        init_array.insert(init_array.len(), dead);
    }
    init_array.insert(init_array.len(), alive);
    print!("{}\n", init_array.join(""));

    for y in 0..automata_length {
        let mut new_array = vec![dead];
        for x in 0..automata_length {
            let analyze = vec![init_array[x], init_array[x + 1], init_array[x + 2]];
            if analyze == vec![dead, dead, dead] {
                new_array.insert(new_array.len(), dead);
            } else if analyze == vec![alive, dead, dead] {
                new_array.insert(new_array.len(), dead);
            } else if analyze == vec![alive, alive, alive] {
                new_array.insert(new_array.len(), dead);
            } else {
                new_array.insert(new_array.len(), alive);
            }
        }
        new_array.insert(new_array.len(), alive);
        init_array = new_array;
        print!("{}\n", init_array.join(""));
    }
}
